File: mobile/lib/shared/models/artist.dart
import 'song.dart';

class Artist {
  final String id;
  final String name;
  final String imageUrl;
  final String? bio;
  final String monthlyListeners;
  final bool isVerified;
  final List<Song> topTracks;

  const Artist({
    required this.id,
    required this.name,
    required this.imageUrl,
    this.bio,
    this.monthlyListeners = '2.4M',
    this.isVerified = true,
    this.topTracks = const [],
  });

  factory Artist.fromJson(Map<String, dynamic> json) {
    return Artist(
      id: json['id']?.toString() ?? '',
      name: json['name']?.toString() ?? 'Unknown Artist',
      imageUrl: json['image']?.toString() ??
          json['imageUrl']?.toString() ??
          '',
      bio: json['bio']?.toString(),
      monthlyListeners: json['monthlyListeners']?.toString() ?? '1.8M',
      isVerified: json['isVerified'] ?? true,
    );
  }
}

File: server/src/models/Artist.js
const mongoose = require('mongoose');

const ArtistSchema = new mongoose.Schema(
  {
    name: {
      type: String,
      required: true,
      unique: true,
      trim: true,
      index: true,
    },
    bio: {
      type: String,
      default: 'Celebrated Tamil Artist on Muxiz.',
    },
    imageUrl: {
      type: String,
      default: '',
    },
    popularTracksCount: {
      type: Number,
      default: 0,
    },
    monthlyListeners: {
      type: Number,
      default: 1000,
    },
    isVerified: {
      type: Boolean,
      default: true,
    },
  },
  {
    timestamps: true,
  }
);

module.exports = mongoose.models.Artist || mongoose.model('Artist', ArtistSchema);
